packages/dom/src/dom/is-text-field.js

```diff
@@ -23,7 +31,9 @@ export default function isTextField( element ) {
 		'number',
 	];
 	return (
-		( nodeName === 'INPUT' && ! nonTextInputs.includes( element.type ) ) ||
+		( isHTMLInputElement( element ) &&
+			element.type &&
+			! nonTextInputs.includes( element.type ) ) ||
 		nodeName === 'TEXTAREA' ||
 		contentEditable === 'true'
```

Comment:
Slight preference for moving the assertion from the destructuring assignment above to here. We should try to keep assertions as small and localized as possible:

```suggestion
		/** @type { HTMLElement } */ ( element ).contentEditable === 'true'
```

Interesting, I just learned that `contentEditable` can also be `"inherit"`. Checking [the `isContentEditable` property](https://developer.mozilla.org/en-US/docs/Web/API/HTMLElement/isContentEditable) is recommended for checking the computed value (in case of `inherit`).

Out of scope for this but worth mentioning 🙂